plugin.video.cbc-sports/addon.py

```diff
@@ -0,0 +1,495 @@
+		params = sys.argv[2]
+		cleanedparams = params.replace('?', '')
+		if (params[len(params) - 1] == '/'):
+			params = params[0:len(params) - 2]
+		pairsofparams = cleanedparams.split('&')
+		param = {}
+		for i in range(len(pairsofparams)):
+			splitparams = {}
+			splitparams = pairsofparams[i].split('=')
+			if (len(splitparams)) == 2:
+				param[splitparams[0]] = splitparams[1]
+
+	return param
+
+
+def addDir(name, url, mode, iconimage, fanart=False, infoLabels=True):
+	u = sys.argv[0] + "?url=" + urllib.parse.quote_plus(url) + "&mode=" + str(mode) + "&name=" + urllib.parse.quote_plus(name) + "&iconimage=" + urllib.parse.quote_plus(iconimage)
+	ok = True
+	liz = xbmcgui.ListItem(name)
+	#liz.setInfo(type="Video", infoLabels={"Title": name})
+	liz.setProperty('IsPlayable', 'true')
+	if not fanart:
+		fanart=defaultfanart
+	liz.setArt({'thumb': iconimage, 'icon': iconimage, 'fanart': fanart})	
+	ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=u, listitem=liz, isFolder=True)
+	return ok
+
+
+def addDir2(name,url,duration,mode,iconimage, aired=False, plot=False, vheight=False, vwidth=False, fanart=False, infoLabels=False):
+	u=sys.argv[0]+"?url="+urllib.parse.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.parse.quote_plus(name)
+	ok=True
+	plot_text = ''
+	if plot:
+	    plot_text = plot
+	else:
+	    plot_text = name
+	aired_text = ''
+	year_test = ''
+	if aired:
+	    aired_text = aired
+	    year_text = aired_text.split('-',1)[0]
+	liz = xbmcgui.ListItem(name)
+	info={ "Title": name,
+             "Aired": aired_text,
+             "Year": aired_text.split('-',1)[0],
+             "mediatype": 'episode',
+             "Duration": duration,
+             "Plot": plot_text }
+	liz.setInfo( "video", info)
+	liz.setProperty('IsPlayable', 'true')
+	fanart=defaultfanart
+	liz.setArt({'thumb': iconimage, 'icon': iconimage, 'fanart': fanart})
+	if vheight and vwidth:
+		aspect = float(float(vwidth) / float(vheight))
+		video_info = {
+			'aspect': aspect,
+			'width': vwidth,
+			'height': vheight,
+			}
+		liz.addStreamInfo('video', video_info)
+	menuitem1 = translation(30011)
+	menuitem2 = translation(30012)
+	liz.addContextMenuItems([ (menuitem1, 'Container.Refresh'), (menuitem2, 'Action(ParentDir)')])
+	ok=xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=u,listitem=liz,isFolder=False)
+	return ok
+
+
+def unescape(s):
+	p = htmllib.HTMLParser(None)
+	p.save_bgn()
+	p.feed(s)
+	return p.save_end()
+
+
+params = get_params()
+url = None
+name = None
+mode = None
+cookie = None
+common.logging_level(cbclog)
+
+try:
+	url = urllib.parse.unquote_plus(params["url"])
+except:
+	pass
+try:
+	name = urllib.parse.unquote_plus(params["name"])
+except:
+	pass
+try:
+	mode = int(params["mode"])
+except:
+	pass
+
+
+if mode == None or url == None or len(url) < 1:
+	CATEGORIES()
+elif mode == 1:
+	INDEX(url)
+elif mode == 2:
+	IFRAME(name,url)
+elif mode == 6:
+	VIDEOS(url)
+elif mode == 7:
+	GET_STREAM(name,url)
+
```

Comment:
usually the addon entrypoints are as small as possible and only pass sys.args elsewhere. This is due to the fact the entrypoint is the only file that's not compiled to bytecode.
Please consider do something like this:

```python
import sys
from resources.lib import myplugin

if __name__ == "__ main__":
  myplugin.run(sys.argv)